### ISSUE TITLE
Support game controllers and joysticks

### ISSUE THREAD

**Issue by ZoogieZork** [HoverRace/HoverRace#205]
Follow-up to #196 (Replace OIS with SDL_Input).

Depends on #198 (Allow mapping of multiple inputs from different "classes" of input to the same action).

The game controller and joystick support was deferred from the OIS-to-SDL migration, since we really need a better way of handling multiple controllers / joysticks, which is covered by #198.
